### ISSUE TITLE
Support renaming tables

### ISSUE THREAD

**Issue by jpsim** [realm/realm-cocoa#2491]
As originally requested here: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/realm-cocoa/sLd7S-jSX40/FeQSRPW0BwAJ

This could be exposed as `-[RLMMigration renameModelFrom:(NSString *)oldName to:(NSString *)newName]`, which would use core's [`Group::rename_table()`](https://github.com/realm/realm-core/blob/4422088866370dfefa70954a7cd103f0da686c5b/src/realm/group.hpp#L341) function.

**Comment by tomlokhorst** [comment]
Is there a workaround to rename a class that works with the current version of RealmSwift?

Based on the ObjC example, I've created this configuration:

```swift
    let config = Realm.Configuration(
      schemaVersion: 2,

      migrationBlock: { migration, oldSchemaVersion in
        if (oldSchemaVersion < 2) {
          migration.enumerate("Person", { (oldObject, newObject) in
            if let oldObject = oldObject {
              print(oldObject)
              migration.create("Employee", value: oldObject)
            }
          })
        }
    })
```

This does print the first `oldObject`, but it crashes on the `migration.create` with the unhelpful error message: `libc++abi.dylib: terminating with uncaught exception of type NSException`

**Comment by jpsim** [comment]
> it crashes on the `migration.create` with the unhelpful error message: `libc++abi.dylib: terminating with uncaught exception of type NSException`

This is only unhelpful if you choose to ignore the exception message. From your update to your comment, it seems like your two classes may have different schemas?

**Comment by tomlokhorst** [comment]
Ah, I didn't know you could catch NSExceptions in Swift, but I [got it to work](http://stackoverflow.com/a/36454808/2597).

This is the error that occurs: `Error Domain=RLMException Code=0 "(null)" UserInfo={RLMRealmVersion=1.1.0, RLMRealmCoreVersion=unknown}`

The only change I made to the schema is renaming the Object.
Person to Employee, in this example.

**Comment by marchy** [comment]
Any plans to support this on the roadmap?

**Comment by ricardopereira** [comment]
@marchy #2499 has been closed. I think this has been delayed.

**Comment by aleksey-ho** [comment]
It's been already implemented.. So simple to be added, but still not merged. :(

**Comment by mefilt** [comment]
TOP

**Comment by marchy** [comment]
So is this available yet or still not merged in?
Been a few years now...

**Comment by mattlewer** [comment]
Another 4 years have passed, is there any update on this feature?

It's as easy in Android as:
schema.rename("A", "B")

Very frustrating we cant do this for iOS

**Comment by aehlke** [comment]
@mattlewer Realm is now dead, there won't be such a feature unless you contribute it yourself

**Comment by mattlewer** [comment]
@aehlke How do you mean it's dead, last release was 2 weeks ago?

**Comment by aehlke** [comment]
@mattlewer it's recently deprecated and will not receive any more commits or releases from parent company later next year. it's transitioning to an open source framework without any parent company supporting it. there's been very little community contribution until now so I don't expect it to become very active after Mongo step down.

**Comment by mattlewer** [comment]
@aehlke I don't believe this to be true. As far as I am aware the Realm atlas sync WITH realm-swift will be deprecated, not realm-swift itself
